### ISSUE TITLE
Parse error with named mapping

### ISSUE THREAD

**Issue by chalex-eth** [Consensys/surya#179]
Hello,

When running some surya function such as inheritance or graph on solidity code (> 0.8.18) that implement mapping we get a parse error.

Let's say the following mapping 

```solidity
mapping(uint256 num1 => uint256 num2) myMapping;
```

Will result in 

```
errors: [
    {
      message: "extraneous input 'num1' expecting '=>'",
      line: 61,
      column: 20
    },
    {
      message: "extraneous input 'num2' expecting ')'",
      line: 61,
      column: 45
    }
  ]
}
```

**Comment by GNSPS** [comment]
Hey @chalex-eth can you please try to uninstall surya and install again?

```
npm uninstall -g surya
npm install -g surya
```

The reason why I suggest this is because there is a floating dependency on the parser and it might just be an old parser version.

**Comment by rashtrakoff** [comment]
> can you please try to uninstall surya and install again?

This doesn't work unfortunately ☹️ . I converted the named mapping to a normal (unnamed) mapping and then it works as expected.

**Comment by vittominacori** [comment]
Same error using OpenZeppelin v5 [ERC20 implementation](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L35) that uses named mapping.

`message: "extraneous input 'account' expecting '=>'",`

**Comment by tintinweb** [comment]
ref: https://github.com/Consensys/vscode-solidity-auditor/issues/137

@GNSPS might require a parser update to v0.16.x

**Comment by GNSPS** [comment]
Updated! 🙏 Closing!

**Comment by hexpaydaydev** [comment]
did not work for me on first install. cleaning cache currently. we'll see if that helps
```

surya@^0.4.6:
  version "0.4.6"
  resolved "https://registry.yarnpkg.com/surya/-/surya-0.4.6.tgz#295a426eff219720d9b7963e04a50206288c321b"
  integrity sha512-zaTYkRbms26cuOWu5jon5l4OsToHX7ZEflqTozXgq/XxUL3VY+tEnxT9Te2WVsA/sYgZPwcH92yQZJgljsss4g==
  dependencies:
    "@solidity-parser/parser" "^0.14.1"
```

**Comment by hexpaydaydev** [comment]
ah, was a new version not published @GNSPS ?

**Comment by hexpaydaydev** [comment]
yup. 2 years ago: https://www.npmjs.com/package/surya

new pkg needs to be published to link it to [solidity metrics](https://github.com/Consensys/solidity-metrics) and beyond

**Comment by GNSPS** [comment]
Yep, sorry, was waiting for a PR fix from myself for `address` member calls.

Published a new "patch" version: v0.4.7, can you try now? 🙏

**Comment by hexpaydaydev** [comment]
works! 🎉
